app/controllers/brapi/v1/germplasm_controller.rb

```diff
@@ -0,0 +1,250 @@
+class Brapi::V1::GermplasmController < Brapi::BaseController
+
+  attr_accessor :request_params, :user
+
+  before_filter :authenticate_api_key!
+
+
+  rescue_from 'ActiveRecord::InvalidForeignKey' do |exception|
+    message = exception.message.split("\n").try(:second)
+    attribute = message ? message[14..-1].split(')')[0] : ''
+    render json: { errors: { attribute: attribute, message: message } }, status: 422
+  end
+
+  rescue_from ActionController::ParameterMissing do |exception|
+    render json: { errors: { attribute: exception.param, message: exception.message } }, status: 422
+  end
+
+
+
+
+ def search
+    # accepted params: germplasmPUI , germplasmDbId , germplasmName , pageSize , page
+    # domainParameters = ['germplasmPUI','germplasmDbId','germplasmName']
+    # totalParameters = domainParameters + ['pageSize','page']
+    
+    if !params['germplasmPUI'].present? && !params['germplasmDbId'].present? && !params['germplasmName'].present?
+      render json: { reason: 'Resource not found' }, status: :not_found
+    else 
+      result_object = germplasm_query(params, false)
+      records = result_object.values
+      
+      
+      if records.nil?
+        render json: { reason: 'Resource not found' }, status: :not_found
+      else
+        if records.size == 1 && records[0]!=nil && records[0] != api_key.user_id && records[1] 
+          render json: { reason: 'This is a private resource of another user' }, status: :unauthorized
+        else
+          json_result_array = []
+          
+          # any programmatic data manipulation can be done here
+          result_object.each do |row|
+            row = JSON.parse(row["row_to_json"])
+            puts row
+            if (!row[:user_id] || (row[:user_id]!=nil && row[:user_id] != api_key.user_id && row[:published] ))
```

Comment:
As BrAPI doesn't currently support ORCID, these lines are being modifying. 
New version will be something like:
if (!row[:user_id] || row[:published] )
as we only should return published datasets, or datasets without an author... if I'm not wrong...